File: CodeWars/8th_Number/sol.cpp
#include <bits/stdc++.h>
using namespace std;

int factor(int x) {
   int ans = 0;
   for (int i = 1; i * i <= x; ++i) {
      if (x % i == 0) {
         ++ans;
         if (i * i != x) ++ans;
      }
   }
   return ans;
}

int main () {
   int n;
   cin >> n;
   int ans = 0;
   for (int i = 1; i <= n; i += 2) {
      if (factor(i) == 8) {
         ans++;
      }
   }
   cout << ans << '\n';
}

File: CodeWars/8th_Number/sol.py
def factor(x):
   ans = 0
   i = 1
   while i * i <= x:
      if x % i == 0:
         ans += 1
         if i * i != x:
            ans += 1
      i += 1
   return ans

ans = 0

n = int(input())

for i in range(1, n + 1, 2):
   if factor(i) == 8:
      ans += 1

print(ans)